drivers/drivers.go

```diff
@@ -0,0 +1,98 @@
+// Package drivers abstracts different object storages, such as local, s3, ipfs
+package drivers
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/net"
+)
+
+// Storages configured at start (used by broadcaster)
+var Storages []OSDriver
+
+// ExternalStorage own external storage, if configured
+var ExternalStorage OSDriver
+
+// LocalStorage single instance of local storage
+var LocalStorage *localOS
+
+// OSDriver common interface for Object Storage
+type OSDriver interface {
+	SaveData(jobID int64, streamID, name string, data []byte) (*net.TypedURI, error)
+	// IsExternal
+	IsExternal() bool
+	// GetInfo return info about object storage need to be able to write this storage
+	GetInfo() *net.OSInfo
+	// EvictManifestData removes memory cache for stream
+	EvictManifestData(mid string)
+}
+
+// AddStorageInstance adds storage driver instance to the list of storages used by broadcaster
+func AddStorageInstance(driver OSDriver) {
+	Storages = append(Storages, driver)
+}
+
+// NewDriver returns new driver
+func NewDriver(storageType net.OSInfo_Storage, info *net.OSInfo) OSDriver {
+	switch storageType {
+	case net.OSInfo_IPFS:
+		return newIPSFDriver()
+	case net.OSInfo_S3:
+		return newS3Driver(info.S3Info)
+	case net.OSInfo_LOCAL:
+		return NewLocalDriver("")
```

Comment:
We don't want this for RPC, because if T provides S3, we don't want B forcing T to use local.

We need a different method for setting the node's own storage; just instantiating local/S3 directly is OK. Will make this change.

Reply:
we want it here, because it is way for B or T to create driver for OS from OSInfo object that other side sent, without knowing concrete type of OS